Library/OcAppleImg4Lib/OcAppleImg4Lib.c

```diff
@@ -0,0 +1,316 @@
+                  NULL,
+                  &DataSize,
+                  &Environment->boardId
+                  );
+  if (EFI_ERROR (Status) || DataSize != sizeof (Environment->boardId)) {
+    Environment->boardId = 0;
+  }
+
+  Environment->certificateEpoch = 2;
+
+  DataSize = sizeof (Environment->securityDomain);
+  Status = gRT->GetVariable (
+                  L"ApSecurityDomain",
+                  &gAppleSecureBootVariableGuid,
+                  NULL,
+                  &DataSize,
+                  &Environment->securityDomain
+                  );
+  if (EFI_ERROR (Status) || DataSize != sizeof (Environment->securityDomain)) {
+    Environment->securityDomain = 1;
+  }
+
+  DataSize = sizeof (Environment->productionStatus);
+  Status = gRT->GetVariable (
+                  L"ApProductionStatus",
+                  &gAppleSecureBootVariableGuid,
+                  NULL,
+                  &DataSize,
+                  &Environment->productionStatus
+                  );
+  if (EFI_ERROR (Status)
+   || DataSize != sizeof (Environment->productionStatus)) {
+    Environment->productionStatus = TRUE;
+  }
+
+  DataSize = sizeof (Environment->securityMode);
+  Status = gRT->GetVariable (
+                  L"ApSecurityMode",
+                  &gAppleSecureBootVariableGuid,
+                  NULL,
+                  &DataSize,
+                  &Environment->securityMode
+                  );
+  if (EFI_ERROR (Status) || DataSize != sizeof (Environment->securityMode)) {
+    Environment->securityMode = TRUE;
+  }
+
+  DataSize = sizeof (Environment->effectiveProductionStatus);
+  Status = gRT->GetVariable (
+                  L"EffectiveProductionStatus",
+                  &gAppleSecureBootVariableGuid,
+                  NULL,
+                  &DataSize,
+                  &Environment->effectiveProductionStatus
+                  );
+  if (EFI_ERROR (Status)
+   || DataSize != sizeof (Environment->effectiveProductionStatus)) {
+    Environment->effectiveProductionStatus = TRUE;
+  }
+
+  DataSize = sizeof (Environment->effectiveSecurityMode);
+  Status = gRT->GetVariable (
+                  L"EffectiveSecurityMode",
+                  &gAppleSecureBootVariableGuid,
+                  NULL,
+                  &DataSize,
+                  &Environment->effectiveSecurityMode
+                  );
+  if (EFI_ERROR (Status)
+   || DataSize != sizeof (Environment->effectiveSecurityMode)) {
+    Environment->effectiveSecurityMode = TRUE;
+  }
+
+  DataSize = sizeof (Environment->internalUseOnlyUnit);
+  Status = gRT->GetVariable (
+                  L"InternalUseOnlyUnit",
+                  &gAppleSecureBootVariableGuid,
+                  NULL,
+                  &DataSize,
+                  &Environment->internalUseOnlyUnit
+                  );
+  if (EFI_ERROR (Status)
+   || DataSize != sizeof (Environment->internalUseOnlyUnit)) {
+    Environment->internalUseOnlyUnit = FALSE;
+  }
+
+  // CHANGE: HardwareModel is unused.
```

Comment:
There is no reason to set it because it is not used anywhere, I'm not sure why it is collected here. I didn't see it in the IMG4 Manifests, the value was part of the range that is UEFI-only control stuff. The value is retrieved and used in OcAppleSecureBootLib to retrieve the correct file. I suppose its collection here is legacy from an older design, possibly it used to be just one Manifest?

Reply:
It makes sense to set and check it to ensure that the manifest was actually designated for this machine and not just renamed in cases where machine certificate is the same across two different models, but it may not be the case here and be really some legacy.

Reply:
Again, I did not see the model *in* the Manifest, but I can check again later.

Reply:
No, the model is *not* part of the Manifest body (checked boot.efi.j132ap.im4m with the ASN.1 JS parser).